src/agents/workflows/subgraphs/step1_preprocessing/graph.py

```
@@ -1,223 +1,238 @@
-    if not images and fp.file_type == "image" and fp.path:
-        images = [fp.path]
+    if not images:
+        paths = _collect_input_paths(state, tool_outputs)
+        images = [p for p in paths if _infer_file_type(p) == "image"]
 
     if not images:
-        print("--- SKIP: NO IMAGES TO ANALYZE ---")
         return state
 
-    provider_cfg = state.get("tool_outputs", {}).get("ocr_provider", {"name": "mock"})
+    tmp_dir_base = Path(tool_outputs.get("tmp_dir", "/tmp/lang_preprocess"))
```

Ruff (0.14.14)

Probable insecure usage of temporary file or directory: "/tmp/lang_preprocess"

(S108)